### ISSUE TITLE
Why this code always print 0?

### ISSUE THREAD

**Issue by iahung** [objeck/objeck-lang#198]
```
class Test {
	function : Main(args : String[]) ~ Nil {
		Brun();
	}

	function : native : Brun() ~ Nil {
		cprev := 3.0;
		cnext := 0.0;
		min := cprev;
		N := 20;
		for(i := 1; i < N; i++) {
			cnext := cprev->Sin();
			if(cnext->Abs() < min) {
				min := cnext->Abs();
			}
			cprev := cnext;
		}
		min->PrintLine();
	}
}
```

**Comment by iahung** [comment]
If I changed `min := cprev;` to `min := 3.0;` then it worked. Why?

**Comment by objeck** [comment]
Thanks, it is a bug in the code optimizer. The code works when compiled as ``obc -src test.obs -opt s0`` but not when compiled as ``obc -src test.obs -opt s1``. The default optimization level is ``s3``.

**Comment by objeck** [comment]
Thanks, it was a bug in the code optimizer's constant propagation routine. The code worked when compiled as ``obc -src test.obs -opt s0`` but not when compiled as ``obc -src test.obs -opt s1``. The default optimization level is ``s3``.

**Comment by objeck** [comment]
Fixed in [7199c9d](https://github.com/objeck/objeck-lang/commit/7199c9d5eb156730e5c4697866405ee2b7dd87f7)

**Comment by objeck** [comment]
@iahung Thank you, I am committed to fixing all compiler and runtime bugs ASAP. Great find!
